test/verify/gemm_softmax_gemm_relu.cpp

```diff
@@ -27,31 +27,48 @@
 #include <migraphx/generate.hpp>
 #include <migraphx/make_op.hpp>
 
-struct gemm_softmax_gemm_relu : verify_program<gemm_softmax_gemm_relu>
+template <bool WithBias, bool WithStandardBiasShape>
+struct gemm_softmax_gemm_relu
+    : verify_program<gemm_softmax_gemm_relu<WithBias, WithStandardBiasShape>>
 {
     migraphx::program create_program() const
     {
         migraphx::program p;
         auto* mm = p.get_main_module();
         migraphx::shape m1_shape{migraphx::shape::half_type, {1, 12, 256, 256}};
-        migraphx::shape m2_shape{migraphx::shape::half_type, {1, 12, 256, 256}};
-        auto m2_elements = m2_shape.elements();
+        auto m2_elements = m1_shape.elements();
         auto a           = mm->add_parameter("1", m1_shape);
         auto b           = mm->add_parameter("2", m1_shape);
         auto b1          = mm->add_parameter("3", m1_shape);
         std::vector<float> eights(m2_elements, 0.125);
-        auto eight = mm->add_literal(migraphx::literal{m2_shape, eights});
-        std::vector<float> zeros(m2_elements, 0);
-        auto zero = mm->add_literal(migraphx::literal{m2_shape, zeros});
+        auto eight = mm->add_literal(migraphx::literal{m1_shape, eights});
 
         b = mm->add_instruction(migraphx::make_op("transpose", {{"permutation", {0, 1, 3, 2}}}), b);
-        auto gemm1   = mm->add_instruction(migraphx::make_op("dot"), a, b);
-        auto scale   = mm->add_instruction(migraphx::make_op("mul"), gemm1, eight);
-        auto bias    = mm->add_instruction(migraphx::make_op("add"), scale, zero);
-        auto softmax = mm->add_instruction(migraphx::make_op("softmax", {{"axis", 3}}), bias);
+        auto gemm1 = mm->add_instruction(migraphx::make_op("dot"), a, b);
+        auto scale = mm->add_instruction(migraphx::make_op("mul"), gemm1, eight);
+
+        std::optional<migraphx::instruction_ref> add_bias{std::nullopt};
+        if constexpr(WithBias)
+        {
+            auto bias_shape = m1_shape;
+            if(not WithStandardBiasShape)
+            {
+                bias_shape = migraphx::shape::from_permutation(
+                    bias_shape.type(), bias_shape.lens(), {0, 1, 3, 2});
+            }
+            auto bias = mm->add_parameter("4", bias_shape);
+            add_bias  = mm->add_instruction(migraphx::make_op("add"), scale, bias);
+        }
+
+        auto softmax = mm->add_instruction(migraphx::make_op("softmax", {{"axis", 3}}),
+                                           WithBias ? add_bias.value() : scale);
         auto gemm2   = mm->add_instruction(migraphx::make_op("dot"), softmax, b1);
         mm->add_instruction(migraphx::make_op("relu"), gemm2);
         return p;
     }
     std::string section() const { return "gemm"; }
 };
+
+template struct gemm_softmax_gemm_relu<false, true>;
```

Comment:
Why is the second parameter set to true?

Reply:
Agree, it looks weird. I addressed this issue with the `enum` that you suggested below.